.test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy

```diff
@@ -38,14 +38,14 @@ mavenJob('beam_PreCommit_Java_MavenInstall') {
   common_job_properties.setMavenConfig(delegate)
 
   // Sets that this is a PreCommit job.
-  common_job_properties.setPreCommit(delegate, 'mvn clean install -pl sdks/java/core -am -amd')
+  common_job_properties.setPreCommit(delegate, 'mvn clean install -pl sdks/java/core,runners/direct-java -am -amd', 'Run Java PreCommit')
 
   // Maven goals for this job: The Java SDK, its dependencies, and things that depend on it.
   goals('''\
     --batch-mode \
     --errors \
     --activate-profiles release,jenkins-precommit,direct-runner,dataflow-runner,spark-runner,flink-runner,apex-runner \
-    --projects sdks/java/core \
+    --projects sdks/java/core,runners/direct-java \
```

Comment:
I don't really remember how 'also-make-dependents' works (and also its strange to me that we would have something that depends on the direct runner but not the core SDK). any insight?

Reply:
`--also-make-dependents` adds everything to the reactor that transitively depends on the SDK. But it does not add their other dependencies so those are pulled from the snapshot repository ;_;

Concrete example:

 - DirectRunner -> Java Core SDK so the direct runner is built
 - DirectRunner -> Fn API

The reactor will include the core SDK and DirectRunner but not the Fn API. This caused a spurious failure on #3924.

What we really want is "Java Core SDK, and all of its dependencies and dependents, with their dependencies pulled from HEAD too" but there is no Maven way to do this that I know of. I briefly considering `-pl '!sdks/python,!sdks/go'` but decided on a whitelist instead.

I think our ultimate path forward will be to have more explicit targets and improve the pipeline precommit, because things like the reference runner and the local artifact stager will depend only on the model. But we cannot build the model with `--also-make-dependents` because that is the whole repo.

Reply:
To be the clear it is the `-pl runners/direct-java --also-make` that pulls in the additional modules I am targeting with this change.